### PR TITLE
Revisit our config for 'Layout' Cops

### PR DESCRIPTION
**Pull request by benthorner**
This updates our config for these Cops to ensure all of
it comes with an explanation. A consequence of this audit
is to enable two Cops:

- Layout/FirstArrayElementIndentation
- Layout/FirstHashElementIndentation

### CODE REVIEW

**Comment by benthorner** [comment]
Update: also tried on Smart Answers, and this matches the predominant code style there too.

File: config/layout.yml [hunk 1]
```diff
@@ -15,8 +15,17 @@ Layout/LineLength:
...
+#
+# SomeClass.first_method_call_on_same_line
+#          .other_method_call
+#
+# a_particularly_long_first_method_call
+#   .subsequent_method_call
+#
+# Any setting of this Cop would cause odd alignment
+# for one of these styles. Since there isn't a Cop to
+# set a preferred style, we should disable this one.
```
Comment (AlanGabbianelli) [inline config/layout.yml:28]: Isn't this consistent with [the `EnforcedStyle: aligned` of `Layout/MultilineMethodCallIndentation`](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Layout/MultilineMethodCallIndentation)?

Comment (benthorner) [reply config/layout.yml:28]: In the first case, yes. The second case is more nuanced than I thought. In our codebases, we have 

```
# before
a_particularly_long_first_method_call
  .subsequent_method_call

# after
a_particularly_long_first_method_call
.subsequent_method_call
```

This is a particularly ugly example/diff for Content Publisher.

```
       expect { described_class.call(params: params, user: user) }
         .to change { edition.reload.summary }.to("New summary")
-        .and change { edition.reload.change_note }.to("New note")
-        .and change { edition.reload.update_type }.to("minor")
+                                             .and change { edition.reload.change_note }.to("New note")
+                                                                                       .and change { edition.reload.update_type }.to("minor")
```

Does that help?

File: config/layout.yml [hunk 2]
```diff
@@ -28,24 +37,42 @@ Layout/AccessModifierIndentation:
...
+#   element_1,
+#   element_2,
+# ]
+#
+# even_in_a_method_call([
+#   element_1,
+#   element_2,
+# ])
+#
 # Introduced in: c69a7eb3af955d6c4c0cf0c3cec8e9f330c74429
```
Comment (kevindew) [inline config/layout.yml:52]: Seems like this should be changed if we're changing the rule

File: config/layout.yml [hunk 3]
```diff
@@ -28,24 +37,42 @@ Layout/AccessModifierIndentation:
...
+#   key_2: value_2,
+# })
+#
 # Introduced in: c69a7eb3af955d6c4c0cf0c3cec8e9f330c74429
-# TODO: unclear why this is here! Suggest enabling this with
-# "EnforcedStyle: consistent".
 Layout/FirstHashElementIndentation:
-  Description: 'Checks the indentation of the first key in a hash literal.'
-  Enabled: false
+  EnforcedStyle: consistent
```
Comment (kevindew) [inline config/layout.yml:70]: Does this cause any problems for code like:

```
even_in_a_method_call(key_1: value_1,
                      key_2: value_2)
```

or:

```
      responses = [{ id: "100000001",
                     submission_time: "2020-04-30T12:00:00+01:00",
                     region: "Greater London",
                     question: "What is my question?",
                     question_format: "In writing, to be read out at the press conference",
                     name: "John Smith",
                     email: "john@example.com",
                     phone: "07123456789" },
                   { id: "100000002",
                     submission_time: "2020-04-30T12:00:00+01:00",
                     region: "Yorkshire and the Humber",
                     question: "What is your question?",
                     question_format: "Record a video, to be shown as part of the press conference",
                     name: "Joe Bloggs",
                     email: "joe@example.com",
                     phone: "07987654321" }]
```

File: config/layout.yml (hunk 2 above)
Comment (benthorner) [reply config/layout.yml]: Sure. I've removed it, as the original purpose was to track the commit messages, which (sometimes) had (more) explanation in them.

File: config/layout.yml (hunk 3 above)
Comment (benthorner) [reply config/layout.yml:66]: [Should do...](https://docs.rubocop.org/en/latest/cops_layout/#layoutfirsthashelementindentation)

> The `consistent` style enforces that the first key in a hash
> literal where the opening brace and the first key are on
> separate lines

I didn't see anything surprising when I trialled it.

Comment (kevindew) [reply config/layout.yml:66]: Great 👍

**Pull request merged**